### PR TITLE
Fixes to send job defined scanning categories in request message

### PR DESCRIPTION
**Pull request by hsong-rh**
https://github.com/ManageIQ/manageiq-providers-amazon/issues/388

### CODE REVIEW

**Comment by hsong-rh** [comment]
@roliveri @jerryk55 Please review.

**Comment by hsong-rh** [comment]
@agrare @bronaghs @roliveri @jerryk55 Please review.

**Comment by hsong-rh** [comment]
This PR is part of the blocker fix for https://bugzilla.redhat.com/show_bug.cgi?id=1533590. Need to be merged ASAP.

File: spec/models/manageiq/providers/amazon/cloud_manager/refresh_parser_spec.rb [hunk 1]
```diff
@@ -4,7 +4,10 @@
 
   context "#get_public_images" do
     it "applies filter from settings.yml" do
-      filter = {:filters => [{:name => "image-type", :values => ["machine"]}]}
+      cfg = Config::Options.new
+      cfg.name = "image-type"
+      cfg.values = ["machine"]
+      filter = {:filters => [cfg]}
```
Comment (agrare) [inline spec/models/manageiq/providers/amazon/cloud_manager/refresh_parser_spec.rb]: You can define the Config::Options inline: `filter = {:filters => [Config::Options.new(:name => "image-type", :values => ["machine"])]}`

Comment (hsong-rh) [reply spec/models/manageiq/providers/amazon/cloud_manager/refresh_parser_spec.rb]: Thanks, Adam. Changed!

**Comment by bronaghs** [comment]
Ah I was just looking into the Travis failure in this repo... thanks for fixing it!
I can speak for the spec test code fix - LGTM.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 1cd2755e67c9a6457180ad732d657a8ad82d6519
Author: Bronagh Sorota <bsorota@redhat.com>
Date:   Thu Jan 18 15:02:47 2018 -0500

    Merge pull request #389 from hsong-rh/scanning_category
    
    Fixes to send job defined scanning categories in request message
    (cherry picked from commit 2abdc7e67cfeaee991a9918856beb0c6dff6a9b0)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1536224
```

**Pull request merged**